hazelcast/src/main/java/com/hazelcast/internal/memory/impl/DirectMemoryBits.java

```diff
@@ -37,7 +41,7 @@ public static char readChar(long address, boolean bigEndian) {
     }
 
     public static char readChar(byte[] buffer, int pos, boolean bigEndian) {
-        return readChar(buffer, (long) (BYTE_ARRAY_BASE_OFFSET + pos), bigEndian);
+        return readChar(BYTE_BUFFER_MEMORY_ACCESSOR, buffer, pos, bigEndian);
```

Comment:
Why do we need to use some unsafe stuff if byte[] is passed explicitly

From my point of view it is ineffective.